examples/common/entry/entry_glfw.cpp

```diff
@@ -46,18 +46,8 @@ namespace entry
 	{
 #	if BX_PLATFORM_LINUX
 # 		if ENTRY_CONFIG_USE_WAYLAND
-		wl_egl_window *win_impl = (wl_egl_window*)glfwGetWindowUserPointer(_window);
-		if(!win_impl)
-		{
-			int width, height;
-			glfwGetWindowSize(_window, &width, &height);
-			struct wl_surface* surface = (struct wl_surface*)glfwGetWaylandWindow(_window);
-			if(!surface)
-				return nullptr;
-			win_impl = wl_egl_window_create(surface, width, height);
-			glfwSetWindowUserPointer(_window, (void*)(uintptr_t)win_impl);
-		}
-		return (void*)(uintptr_t)win_impl;
+		struct wl_surface* surface = (struct wl_surface*)glfwGetWaylandWindow(_window);
+		return (void*)surface;
```

Comment:
Code there was added by @BlockoS, and it looks like he had some reason why it was more complex than just `glfwGetWaylandWindow`.

Reply:
Yeah, the main reason it should be in EGL is to link it up with the resizing events that need to get forwarded to `wl_egl_window_resize`.